packages/layer-composer/src/types/index.ts

```diff
@@ -1,5 +1,18 @@
 import type { Layer, AnySourceImpl, Style } from 'mapbox-gl'
-import { GeneratorConfig, Type, GeneratorLegend } from '../generators/types'
+import type { DataviewConfig } from '@globalfishingwatch/api-types'
+import {
+  GeneratorConfig,
+  Type,
+  GeneratorLegend,
+  HeatmapAnimatedGeneratorSublayer,
+  ColorRampsIds,
+} from '../generators/types'
+
+export interface GeneratorDataviewConfig extends DataviewConfig<Type> {
```

Comment:
I'm mot super clear why this type is needed, but isn't it a bit confusing? If we need a `HeatmapAnimatedConfig` or a `BasemapConfig`, wouldn't it be better to cast as one of these types rather than having this weird generic type?

Reply:
You are right, will be nice to have the option of typing by each different one and we'll use the one that composes all of them as fallback
So it will be Type by default but in the instance, we could pass Heatmap or Basemap or whatever we need.
```suggestion
export interface GeneratorDataviewConfig extends DataviewConfig<T = Type>
```